core/common/src/main/java/alluxio/grpc/GrpcChannel.java

```diff
@@ -51,6 +52,11 @@ public GrpcChannel(GrpcManagedChannelPool.ChannelKey channelKey, Channel channel
     mChannelHealthState = channel instanceof AuthenticatedChannel
         ? () -> (((AuthenticatedChannel) channel).isAuthenticated() && mChannelHealthy)
             : () -> mChannelHealthy;
+    if (channel instanceof AuthenticatedChannel) {
```

Comment:
we do various things for the instance of `AuthenticatedChannel`. Can we just have a bool for that?

Also, what other types of channels (not AuthenticatedChannel) are possible? All this checking looks like it can be abstracted away.

Reply:
`Channel` is not our class so can't replace `AuthenticatedChannel` with a boolean. I did some changes for better abstracting the authenticated channel handling.